README.md

```diff
@@ -4619,6 +4632,301 @@ The command above assumes you mounted the data workspace in `/mount/data`, and t
+    model.rlhf.critic.port=${CRITIC_PORT}
+EOF
+
+srun --het-group=3 --container-image=${CONTAINER} bash -c "${cmd_ppo}" &
+
+# END HETEROGENEUS JOB 3
+
+wait
+```
+It is important to launch each job with & after the srun command, to ensure each job doesn’t block the next one. The wait statement at the end of script ensures that the entire job does not exit until each individual job is finished.
+
+Note: the three servers do not support data parallelism. Therefore, the SLURM –ntasks-per-node value should be set to the model parallelism value (tensor parallelism * pipeline parallelism) for that same job. And the trainer.devices value must also be set to that same value as well. However, the PPO actor supports data parallelism, so –ntasks-per-node can be set to the number of GPUs in each node.
+
+##### 5.14.2.6 PPO Hyper-parameters
+<a id="markdown-ppo-hyper-parameters" name="ppo-hyper-parameters"></a>
+
+All the model related parameters can be controlled the same way as in other NeMo training jobs. However, we also provide full control of the behavior of PPO during training, with a section in the config yaml files inside model.rlhf. These are the descriptions of the available hyper-parameters:
+
+- Rlhf.reward_model: Provide the ip address and the port where the Reward Model will be running, to enable communication with it.
+- Rlhf.critic: Provide the ip address and the port where the PPO Critic will be running, to enable communication with it.
+- Rlhf.initial_policy: Provide the ip address and the port where the Initial Policy will be running, to enable communication with it.
+- Rlhf.ppo.entropy_penalty: Control the effect of the entropy term in PPO.
+- Rlhf.ppo.inital_pollicy_kl_penalty: Control the effect of the initial policy KL Divergence term in PPO.
+- Rlhf.ppo.epochs: Number of epochs the actor and critic will perform on the data stored in the rollout buffer each time.
+- Rlhf.ppo.num_rollout_samples: Number of samples that will be generated during the rollout stage before moving to the training stage.
+- Rlhf.ppo.rollout_micro_batch_size: Micro batch size for the rollout phase. Each GPU will load this many prompts and generate responses for them.
+- Rlhf.ppo.ratio_eps: epsilon value for clipping the PPO ratio during training.
+- Rlhf.ppo.discount: discount factor for calculating the returns and advantages.
+- Rlhf.ppo.gae_lambda: lambda value for the Generalized Advantage Estimation (GAE) calculation.
+- Rlhf.ppo.normalize_advantage: whether to normalize the advantages to have a mean of zero and standard deviation of one.
+
+During the rollout phase, the sampling parameters for the model can also be modified, by using the parameters in model.sampling_params.
+
+#### 5.14.3. Future Work
+<a id="markdown-future-work" name="future-work"></a>
+
+- Our reward model currently supports only datasets with two responses per prompt. We will add support for training with datasets that have more than 1 comparison in future releases.
```

Comment:
Consider rephrasing, e.g. ... two responses per prompt ... We will add support for training with datasets that have more than two responses per prompt in future releases.

Reply:
https://github.com/NVIDIA/NeMo-Megatron-Launcher/pull/30/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R4927